src/pudl/transform/ferc1.py

```diff
@@ -4840,10 +4841,101 @@ class UtilityPlantSummaryFerc1TableTransformer(Ferc1AbstractTableTransformer):
     table_id: TableIdFerc1 = TableIdFerc1.UTILITY_PLANT_SUMMARY_FERC1
     has_unique_record_ids: bool = False
 
+    def process_xbrl_metadata(self, xbrl_metadata_json) -> pd.DataFrame:
+        """Do the default metadata processing plus add a new factoid.
+
+        The new factoid cooresponds to the aggregated factoid in
+        :meth:`aggregated_xbrl_factoids`.
+        """
+        tbl_meta = super().process_xbrl_metadata(xbrl_metadata_json)
+        # things that could be grabbed from a aggregated_xbrl_factoids param
+        factoids_to_agg = [
+            "utility_plant_in_service_classified",
+            "utility_plant_in_service_property_under_capital_leases",
+        ]
+        new_factoid_name = (
+            "utility_plant_in_service_classified_and_property_under_capital_leases"
+        )
+
+        calc = [
+            {"name": factoid, "weight": 1, "source_tables": [self.table_id.value]}
+            for factoid in factoids_to_agg
+        ]
+        new_fact = pd.read_csv(
+            StringIO(
+                f"""xbrl_factoid,calculations,balance,ferc_account,xbrl_factoid_original,intra_table_calc_flag,row_type_xbrl
+{new_factoid_name},[],debit,,{new_factoid_name},True,calculated_value
+"""
+            )
+        ).convert_dtypes()
+        # I couldn't figure out how to format the csv version above to get it in the
+        # first time so i'm adding it down here.
+        new_fact.loc[0, "calculations"] = json.dumps(calc)
```

Comment:
This section feels hard to read to me. This seems to yield equivalent results:

```py
calc = [
    {
        "name": factoid,
        "weight": 1.0,
        "source_tables": [self.table_id.value],
    }
    for factoid in factoids_to_agg
]

new_fact = pd.DataFrame({
    "xbrl_factoid": [new_factoid_name],
    "calculations": [json.dumps(calc2)],
    "balance": ["debit"],
    "ferc_account": [pd.NA],
    "xbrl_factoid_original": [new_factoid_name],
    "intra_table_calc_flag": [True],
    "row_type_xbrl": ["calculated_value"],
}).convert_dtypes()
```

Reply:
yeah this looks way nicer ty for the suggestion